presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -421,6 +421,21 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     }
 
     private ConnectorTableMetadata getTableMetadata(SchemaTableName tableName)
+    {
+        try {
+            return doGetTableMetadata(tableName);
+        }
+        catch (PrestoException e) {
+            throw e;
+        }
+        catch (RuntimeException e) {
+            // Errors related to invalid or unsupported information in the Metastore should be handled explicitly (eg. as PrestoException(HIVE_INVALID_METADATA)).
+            // This is just a catch-all solution so that we have any actionable information when eg. SELECT * FROM information_schema.columns fails.
+            throw new RuntimeException("Failed to construct table metadata for table " + tableName, e);
+        }
```

Comment:
```
        catch (RuntimeException e) {
            propagateIfInstanceOf(PrestoException.class);
            // Errors related to invalid or unsupported information in the Metastore should be handled explicitly (eg. as PrestoException(HIVE_INVALID_METADATA)).
            // This is just a catch-all solution so that we have any actionable information when eg. SELECT * FROM information_schema.columns fails.
            throw new RuntimeException("Failed to construct table metadata for table " + tableName, e);
        }
```
?

Reply:
Should we throw here some "generic" PrestoException (that way no one else with wrap it again).

Reply:
no, because this is not a proper end-user error reporting. This is a catch-all only